x-pack/libbeat/management/error.go

```diff
@@ -0,0 +1,108 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package management
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/gofrs/uuid"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/x-pack/libbeat/management/api"
+)
+
+// ErrorType is type of error that the events endpoint understand.
+type ErrorType string
+
+// ConfigError is the type of error send when an unpack or a blacklist happen.
+var ConfigError = ErrorType("CONFIG")
+
+// ErrorEvent is the event type when an error happen.
+var ErrorEvent = api.EventType("ERROR")
+
+// Error is a config error to be reported to kibana.
+type Error struct {
+	Type ErrorType
+	UUID uuid.UUID
+	Err  error
+}
```

Comment:
I think I will change that and make `Err` just a string instead it will be more appropriate.

Reply:
If you turn it into a string, maybe rename the field to 'Message' ?